### PR TITLE
Feature/simd power

### PR DESCRIPTION
**Pull request by gMerm**
Used Altivec Intrinsics in order to calculate average

### CODE REVIEW

File: average_power.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    // load the corresponding 4 values from the loop, perform addition & store
+    // result in vec
+    vector float input = vec_ld(0, &A[i]);
+    vec = vec_add(vec, input);
+  }
+
+  // add all the 4 parts of the vector and save them in sum vector
+  float sum_array[4];
+  vec_st(vec, 0, sum_array);
+  avg2 = sum_array[0] + sum_array[1] + sum_array[2] + sum_array[3];
```
Comment (markos) [inline average_power.c]: there is a way to do it in VSX before the vec_st, in essence it's this:
// vec = | V1 | V2 | V3 | V4 |
tmp = vec_sld(vec, vec, 8); // shift vector 8 bytes to the left tmp | V3 | V4 | V1 | V2 |
tmp = vec_add(vec, tmp); // add vec + tmp -> | V3 + V1 | V4 + V2 | V1 + V3 | V2 + V4 |
tmp2 = vec_sld(tmp,tmp, 4); // tmp2: | V4 + V2 | V1 + V3 | V2 + V4 | V3 + V1 |
vec = vec_add(tmp, tmp2); // vec = | V3 + V1 + V4 + V2 | ...
vec_st(...) // all elements have the same value
// or vec_ste() single element 0
return sum_array[0].

**Comment by markos** [comment]
@gMerm please resolve the conflicts in the Makefile so that it can be merged.

**Comment by gMerm** [comment]
> @gMerm please resolve the conflicts in the Makefile so that it can be merged.

done.

**Comment by markos** [comment]
```
$ make
Makefile:31: *** missing separator.  Stop.
```

**Pull request closed without merging**